pagopa/io-app
# feat: [IOCOM-643] Back to SEND message screen upon successful payment completion

## Short description
This PR changes the payment flow in order to go back to the SEND message screen when a payment is completed successfully.
|Pick a payment|Complete the payment|Redirected to the SEND message|
|-|-|-|
|![Simulator Screenshot - iPhone 14 - 2023-10-31 at 16 35 09](https://github.com/pagopa/io-app/assets/5150343/2487a4bf-4339-4263-89ef-d5463d3cd15c)|![Simulator Screenshot - iPhone 14 - 2023-10-31 at 16 35 26](https://github.com/pagopa/io-app/assets/5150343/d127eb3b-898c-4f63-a261-c81b5623640a)|![Simulator Screenshot - iPhone 14 - 2023-10-31 at 16 35 59](https://github.com/pagopa/io-app/assets/5150343/1b599cf7-547f-4b28-8617-27172c9e94ed)|

## List of changes proposed in this pull request
- Added a new property `shouldGoBackToEntrypointOnSuccess` to `mapStateToProps` in `PaymentOutcomeCodeMessage`
- Such property is used to choose what to do after the user acknowledges a successful payment
  - If `true`, `backToEntrypointPayment` action is dispatched (and handled by the existing saga)
  - else, the wallet home is displayed (as before)
- Fixed an unrelated typo

## How to test
Using _io-dev-api-server_ configured with a SEND message with payments, do a successful payment and check that, upon completion, you are redirected back to the SEND message
